packages/process/src/node/multi-ring-buffer.ts

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+import * as stream from 'stream';
+import { inject, injectable } from 'inversify';
+import { Disposable } from '@theia/core/lib/common';
+
+export class MultiRingBufferReadableStream extends stream.Readable implements Disposable {
+
+    protected more = false;
+
+    constructor(protected readonly ringBuffer: MultiRingBuffer,
+        protected readonly reader: number,
+        protected readonly encoding = 'utf8'
+    ) {
+        super();
+        this.setEncoding(encoding);
+    }
+
+    _read(size: number) {
+        this.more = true;
+        this.deq(size);
+    }
+
+    onData() {
+        if (this.more === true) {
+            this.deq(-1);
+        }
+    }
+
+    deq(size: number) {
+        let buffer = undefined;
+        do {
+            buffer = this.ringBuffer.deq(this.reader, size, this.encoding);
+            if (buffer !== undefined) {
+                this.more = this.push(buffer, this.encoding);
+            }
+        }
+        while (buffer !== undefined && this.more === true);
+    }
+
+    dispose() {
+        this.ringBuffer.closeStream(this);
+        this.ringBuffer.closeReader(this.reader);
+    }
+}
+
+export const MultiRingBufferOptions = Symbol('MultiRingBufferOptions');
+export interface MultiRingBufferOptions {
+    readonly size: number,
+    readonly encoding?: string,
+}
+
+export interface WrapedPosition { newPos: number, wrap: boolean }
+
+@injectable()
+export class MultiRingBuffer {
```

Comment:
The `MultiRingBuffer` seems to be a bit involved but easy to write tests against. Could you add some?

Reply:
Yes tests are in multi-ring-buffer.spec.ts ? code coverage is at 98%
So you see any missing ?  
Github doesn't show the diff because it's too bid.. maybe you missed it ?

Reply:
Great. I missed them indeed. :)